apps/api/bronya.config.ts

```diff
@@ -141,25 +160,99 @@ class MyStack extends Stack {
 }
 
 export async function main() {
+  const id = "peterportal-api-next";
+
   const app = new App();
 
-  const stack = new MyStack(app, "peterportal-api-canary");
+  if (!process.env.NODE_ENV) {
+    throw new Error("NODE_ENV not set.");
+  }
+
+  let stage;
+
+  switch (process.env.NODE_ENV) {
+    case "production":
+      stage = "prod";
+      break;
+    case "staging":
+      if (!process.env.PR_NUM) {
+        throw new Error("NODE_ENV was set to staging, but a PR number was not provided.");
+      }
+      stage = `staging-${process.env.PR_NUM}`;
+      break;
+    case "development":
+      stage = "dev";
+      break;
+    default:
+      throw new Error(
+        "Invalid NODE_ENV specified. Valid values are 'production', 'staging', and 'development'.",
+      );
+  }
```

Comment:
Can we encapsulate this as a function defined at the top level of the module (or maybe in `@libs/common` :eyes:)

```suggestion
// at the top-level
function getStage() {
  switch (process.env.NODE_ENV) {
    case "production":
      return "prod";
    case "staging":
      if (!process.env.PR_NUM) {
        throw new Error("NODE_ENV was set to staging, but a PR number was not provided.");
      }
      return `staging-${process.env.PR_NUM}`;
    case "development":
      return "dev";
    default:
      throw new Error(
        "Invalid NODE_ENV specified. Valid values are 'production', 'staging', and 'development'.",
      );
  }
}

const stage = getStage()

```